offline.py

```diff
@@ -0,0 +1,201 @@
+"""Verification function to plot predictions and target data."""
+
+# Standard library
+import gc
+import os
+from argparse import ArgumentParser
+
+# Third-party
+import cartopy.feature as cf
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import torch
+import xarray as xr
+
+# First-party
+from neural_lam import constants, utils
+from neural_lam.weather_dataset import WeatherDataModule
+
+
+@matplotlib.rc_context(utils.fractional_plot_bundle(1))
+def offline_plotting():
+    """Plot expected output against npy prediction
+    generated by the model.
+    """
+    parser = ArgumentParser(
+        description="Standalone offline plotting of zarr and npy files"
+    )
+
+    parser.add_argument(
+        "--path_target_file",
+        type=str,
+        default="",
+        help="Path to the .zarr archive to verify against - target",
+    )
+    parser.add_argument(
+        "--path_prediction_file",
+        type=str,
+        default="",
+        help="Path to the file output from the inference as .npy",
+    )
+    parser.add_argument(
+        "--saving_path",
+        type=str,
+        default="",
+        help="Path to save the graphical output of this function",
+    )
+    parser.add_argument(
+        "--variable_to_plot",
+        type=str,
+        default="TQV",
+        help="Variable to plot in short format",
+    )
+
+    # Get args
+    args = parser.parse_args()
+
+    # # Mapping out the feature channel to its index
+    mapping_dictionary = precompute_variable_indices()
+    feature_channel = mapping_dictionary[args.variable_to_plot][0]
+
+    # Load inference dataset
+    predictions_data_module = WeatherDataModule(
+        "cosmo_old",
+        path_verif_file=args.path_prediction_file,
+        standardize=False,
+        subset=False,
+        batch_size=6,
+        num_workers=2,
+    )
+    predictions_data_module.setup(stage="verif")
+    predictions_loader = predictions_data_module.verif_dataloader()
+    for predictions_batch in predictions_loader:
+        predictions = predictions_batch[0]
+        break
+
+    # Space on memory
+    del predictions_data_module, predictions_loader
+    gc.collect()
+
+    # Output the prediction time range
+    start_time_str = constants.EVAL_DATETIMES[0]  # This should be '2015112800'
+    start_time = pd.to_datetime(start_time_str, format="%Y%m%d%H")
+
+    # Output the prediction time range
+    time_range = len(predictions[1, :, 1, 1])  # Number of time steps
```

Comment:
```suggestion
    time_range = len(predictions[0, :, 0, 0])  # Number of time steps
```
use first indices to avoid issues with edge cases